src/number_format.js

```diff
@@ -644,11 +637,38 @@ class NumberFormat extends React.Component {
       return value;
     }
 
-    //if format got deleted reset the value to last value
-    if (this.checkIfFormatGotDeleted(start, end, lastValue)) {
-      value = lastValue;
+    // check whether the deleted portion has a non numeric field
+    // this means that it may be a format pattern symbol or a separator
+    const deletedValues = lastValue.substr(start, end - start);
+    const hasNonNumeric = !![...deletedValues].find((deletedVal, idx) => this.isCharacterAFormat(idx + start, lastValue));
+
+    // if it has, only remove the numeric portion
+    if(hasNonNumeric) {
+      const deletedValuePortion = lastValue.substr(start)
+      const recordIndexOfFormatCharacters = {};
+      const resolvedPortion = [];
+      [...deletedValuePortion].forEach((currentPortion, idx) => {
+        if(this.isCharacterAFormat(idx + start, lastValue)){
+          recordIndexOfFormatCharacters[idx] = currentPortion;
+        } else if (idx > deletedValues.length - 1) {
+          resolvedPortion.push(currentPortion);
+        }
+      })
+
+      Object.keys(recordIndexOfFormatCharacters).forEach(idx => {
+        if(resolvedPortion.length > idx){
+          resolvedPortion.splice(idx, 0, recordIndexOfFormatCharacters[idx]);
+        } else {
+          resolvedPortion.push(recordIndexOfFormatCharacters[idx])
+        }
+      })
+
+      value = lastValue.substr(0, start) + resolvedPortion.join('');
```

Comment:
The problem is the format character can have numeric value in itself.
This logic makes the assumption that there will be no numeric char on format itself.

Reply:
My comment was misleading.  I wasn't checking if the characters that were removed were numeric.  I corrected it and also create unit test to back it up